duckduckgo/iOS
# Autofill logins

Task/Issue URL: https://app.asana.com/0/72649045549333/1201600439227198/f
Tech Design URL:
CC:

**Description**:
Adds V1 of autofilling logins to iOS. This includes filling and saving (including partial credentials), and some of the settings interface. There's a comprehensive list of what should be in this here: https://app.asana.com/0/0/1202346026470299/f
And a list of things deferred until V2 here: https://app.asana.com/0/0/1202345981744059/f

Also some design review items deferred if you spot any of those: https://app.asana.com/0/0/1202374896817644/f

Also worth noting that there are strings that deliberately aren't in UserText as they shouldn't be translated yet given the internal nature of this release



**Steps to test this PR**:
----
Test email autofill (this shouldn't have changed, so you can just briefly check it's still there)
1. Sign out of email if signed in
1. Test that email autofill does not appear 
1. Sign in to email. 
1. Test email autofill still works (both duck address and generated aliases, test the generated ones are different every time).
----
Test saving logins
Test using both a login and a sign up form. I use https://www.katesclothing.co.uk/account/login and https://www.katesclothing.co.uk/account/register
1. On the first save, you should see a different screen introducing you to the idea of autofill.
1. Test saving one with a long first part of the email (e.g. loooooooooooooooooooooooooooongemail@example.com), when displaying this email it should truncate just the first part (e.g.. looooooo...@example.com)
1. Test only typing a password and saving. It should save only the password
1. Now type the same password, but with a username, it should offer to update (and end up saving as one set of credentials).
1. Now type the same username, but change the password. It should prompt to update the password
----
1. Test Autofilling logins
I use https://www.katesclothing.co.uk/account/login to test
Test saving greater than 3 on the same site, then the autofilling logins modal should have a show more button



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
